### PR TITLE
Add extendable API 2.0  for OpenVINO

### PR DESCRIPTION
**Pull request by preetha-intel**
### Description
Add extendible API for OpenVINO - 2.0

### CODE REVIEW

File: include/onnxruntime/core/providers/openvino/openvino_provider_options.h [hunk 1]
```diff
@@ -0,0 +1,26 @@
...
+/// Going forward, new OpenVINO provider options are to be supported via this struct and usage of the publicly defined
+/// OrtOpenVINOProviderOptions will be deprecated over time.
+/// </summary>
+struct OrtOpenVINOProviderOptionsV2 {
+  /** \brief Device type string
+   *
+   * Valid settings are one of: "CPU_FP32", "CPU_FP16", "GPU_FP32", "GPU_FP16"
+   */
+  const char* device_type;
+  unsigned char enable_vpu_fast_compile;   ///< 0 = disabled, nonzero = enabled
```
Comment (sfatimar) [inline include/onnxruntime/core/providers/openvino/openvino_provider_options.h:19]: Maybe we can remove vpu_fast_compile

**Review by sfatimar — COMMENTED** [review]
Please raise a PR with MSFT for them to review this separately as it is API Changes

File: onnxruntime/python/onnxruntime_pybind_state.cc [hunk 2]
```diff
@@ -593,7 +594,7 @@ std::unique_ptr<IExecutionProvider> CreateExecutionProviderInstance(
 #endif
   } else if (type == kOpenVINOExecutionProvider) {
 #ifdef USE_OPENVINO
-    OrtOpenVINOProviderOptions params;
+    OrtOpenVINOProviderOptionsV2 params;
```
Comment (sfatimar) [inline onnxruntime/python/onnxruntime_pybind_state.cc:597]: Will changing the structure here impact backward compatibility for example if you remove vpu_fast_compile

File: onnxruntime/python/onnxruntime_pybind_state_common.h [hunk 3]
```diff
@@ -448,7 +449,7 @@ std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_MIGrap
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_MIGraphX(int device_id);
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Cuda(const OrtCUDAProviderOptions* params);
 std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_Dnnl(const OrtDnnlProviderOptions* params);
-std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_OpenVINO(const OrtOpenVINOProviderOptions* params);
+std::shared_ptr<IExecutionProviderFactory> CreateExecutionProviderFactory_OpenVINO(const OrtOpenVINOProviderOptionsV2* params);
 #ifdef USE_TVM
```
Comment (sfatimar) [inline onnxruntime/python/onnxruntime_pybind_state_common.h:453]: same comment as above

File: onnxruntime/test/perftest/ort_test_session.cc [hunk 4]
```diff
@@ -426,15 +427,15 @@ OnnxRuntimeTestSession::OnnxRuntimeTestSession(Ort::Env& env, std::random_device
         ORT_THROW("[ERROR] [OpenVINO] wrong key type entered. Choose from the following runtime key options that are available for OpenVINO. ['device_type', 'device_id', 'enable_vpu_fast_compile', 'num_of_threads', 'cache_dir', 'enable_opencl_throttling|true'] \n");
       }
     }
-    OrtOpenVINOProviderOptions options;
+    OrtOpenVINOProviderOptionsV2 options;
```
Comment (sfatimar) [inline onnxruntime/test/perftest/ort_test_session.cc:430]: same comment what happens when we remove vpu_fast_compile from the struct

**Comment by preetha-intel** [comment]
Moving to common ProviderOptions from ORT. Closing this new API as its no longer relevant.

**Pull request closed without merging**